src/main/java/org/stocker/services/StockDBClientImpl.java

```diff
@@ -31,7 +31,12 @@ public boolean insert(NseDataObj dataObj){
     }
 
     @Override
-    public JsonObject retrieve(@NotNull String symbol) {
+    public JsonObject retrieve(String symbol) {
+        if (Strings.isNullOrEmpty(symbol)) {
+            // symbol should never be empty or null
+            return null;
```

Comment:
never have APIs returning nulls . Return empty JSONObjects if necessary . Otherwise everytime we use retrieve , we'll end up having a if( res == null ) block .